frontend/src/modules/placement/reducers.ts

```diff
@@ -442,30 +411,42 @@ function placementDestinationLocations(state: PlacementState, sources: Draft<Cel
             break
         }
         case PlacementType.GROUP: {
-            // it is possible to place samples from multiple containers in one shot
-
-            // sort source location indices by sample id
-            const sourceIndices = [...sources.keys()].sort((indexA, indexB) => {
+            const relativeOffsetByIndices = [...sources.keys()].sort((indexA, indexB) => {
                 const a = sources[indexA]
                 const b = sources[indexB]
-                const sampleA = a.sample
-                const sampleB = b.sample
-                if (sampleA === null) {
-                    throw new Error(`Cell at coordinates ${atCellLocations(a)} has no sample`)
-                }
-                if (sampleB === null) {
-                    throw new Error(`Cell at coordinates ${atCellLocations(b)} has no sample`)
+                const offsetsA = a.coordinates ? coordinatesToOffsets(getContainer(state, a).spec, a.coordinates) : []
```

Comment:
Might be worth extracting that offset comparison as a function seeing as it is recurring...